### DISCUSSION TITLE
What exact arguments do I need to run sd_api_picture?

### DISCUSSION THREAD

**Discussion by gsgoldma** [oobabooga/text-generation-webui#563] [Q&A]
I know both auto1111 SD and Oobabooga text-ui both require some --flag arguments but I'm not sure exactly which one's for each.
I've gotten it to work briefly, but it only flashed an image for a moment. Most of the time it just hangs and nothing sends or is even generated.

It actually generated something one time, but the second time generates nothing and hangs.

API error: POST: http://127.0.0.1:7860/sdapi/v1/txt2img {'error': 'IndexError', 'detail': '', 'body': '', 'errors': 'list index out of range'}
╭───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────╮
│ D:\stable-diffusion-webui\venv\lib\site-packages\anyio\streams\memory.py:94 in receive                               │
│                                                                                                                      │
│ D:\stable-diffusion-webui\venv\lib\site-packages\anyio\streams\memory.py:89 in receive_nowait                        │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
WouldBlock

During handling of the above exception, another exception occurred:

╭───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────╮
│ D:\stable-diffusion-webui\venv\lib\site-packages\starlette\middleware\base.py:77 in call_next                        │
│                                                                                                                      │
│ D:\stable-diffusion-webui\venv\lib\site-packages\anyio\streams\memory.py:114 in receive                              │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
EndOfStream

During handling of the above exception, another exception occurred:

╭───────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────╮
│ D:\stable-diffusion-webui\modules\api\api.py:144 in exception_handling                                               │
│                                                                                                                      │
│   143 │   │   try:                                                                                                   │
│ ❱ 144 │   │   │   return await call_next(request)                                                                    │
│   145 │   │   except Exception as e:                                                                                 │
│                                                                                                                      │
│ ╭────────────────────────────────────────────── locals ──────────────────────────────────────────────╮               │
│ │        call_next = <function BaseHTTPMiddleware.__call__.<locals>.call_next at 0x0000028BC3A2BF40> │               │
│ │                e = IndexError('list index out of range')                                           │               │
│ │ handle_exception = <function api_middleware.<locals>.handle_exception at 0x0000028BC38FB400>       │               │
│ │          request = <starlette.requests.Request object at 0x0000028BD2189180>                       │               │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────────╯               │
│                                                                                                                      │
│ D:\stable-diffusion-webui\venv\lib\site-packages\starlette\middleware\base.py:80 in call_next                        │
│                                                                                                                      │
│                                               ... 27 frames hidden ...                                               │
│                                                                                                                      │
│ D:\stable-diffusion-webui\scripts\prompt_blending.py:39 in <lambda>                                                  │
│                                                                                                                      │
│    38 │                                                                                                              │
│ ❱  39 │   switched_prompts = list(map(lambda p: switch_syntax(p), prompts))                                          │
│    40 │   return real_get_learned_conditioning(model, switched_prompts, steps)                                       │
│                                                                                                                      │
│ ╭────────────────────────────────────────── locals ──────────────────────────────────────────╮                       │
│ │ p = '(Masterpiece:1.1), (solo:1.3), detailed, intricate, colorfulI can help you with '+512 │                       │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────╯                       │
│                                                                                                                      │
│ D:\stable-diffusion-webui\scripts\prompt_blending.py:157 in switch_syntax                                            │
│                                                                                                                      │
│   156 │   │                                                                                                          │
│ ❱ 157 │   │   if c == REAL_MARK and stack[-1] == '{':                                                                │
│   158 │   │   │   p[i] = MARK                                                                                        │
│                                                                                                                      │
│ ╭──────────────────────────────────────────── locals ─────────────────────────────────────────────╮                  │
│ │      c = ':'                                                                                    │                  │
│ │      i = 256                                                                                    │                  │
│ │      p = ['(', 'M', 'a', 's', 't', 'e', 'r', 'p', 'i', 'e', ... +582]                           │                  │
│ │ prompt = '(Masterpiece:1.1), (solo:1.3), detailed, intricate, colorfulI can help you with '+512 │                  │
│ │  stack = []                                                                                     │                  │
│ ╰─────────────────────────────────────────────────────────────────────────────────────────────────╯                  │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
IndexError: list index out of range

saw this on auto1111 webui afterwards

Traceback (most recent call last):
  File "D:\oobabooga\installer_files\env\lib\site-packages\gradio\routes.py", line 374, in run_predict
    output = await app.get_blocks().process_api(
  File "D:\oobabooga\installer_files\env\lib\site-packages\gradio\blocks.py", line 1017, in process_api
    result = await self.call_function(
  File "D:\oobabooga\installer_files\env\lib\site-packages\gradio\blocks.py", line 849, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "D:\oobabooga\installer_files\env\lib\site-packages\anyio\to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "D:\oobabooga\installer_files\env\lib\site-packages\anyio\_backends\_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "D:\oobabooga\installer_files\env\lib\site-packages\anyio\_backends\_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "D:\oobabooga\installer_files\env\lib\site-packages\gradio\utils.py", line 453, in async_iteration
    return next(iterator)
  File "D:\oobabooga\text-generation-webui\modules\chat.py", line 177, in cai_chatbot_wrapper
    for _history in chatbot_wrapper(text, max_new_tokens, do_sample, temperature, top_p, typical_p, repetition_penalty, encoder_repetition_penalty, top_k, min_length, no_repeat_ngram_size, num_beams, penalty_alpha, length_penalty, early_stopping, seed, name1, name2, context, check, chat_prompt_size, chat_generation_attempts):
  File "D:\oobabooga\text-generation-webui\modules\chat.py", line 128, in chatbot_wrapper
    visible_reply = apply_extensions(visible_reply, "output")
  File "D:\oobabooga\text-generation-webui\modules\extensions.py", line 37, in apply_extensions
    text = extension.output_modifier(text)
  File "D:\oobabooga\text-generation-webui\extensions\sd_api_pictures\script.py", line 125, in output_modifier
    image = get_SD_pictures(string)
  File "D:\oobabooga\text-generation-webui\extensions\sd_api_pictures\script.py", line 85, in get_SD_pictures
    for img_str in r['images']:
KeyError: 'images'

and the above was in oobabooga webui

Edit: Tried again and it worked twice in a row, but it just took a very long time. It siad 400 seconds, but I think it took much longer. There is also no indicator that it's working until it starts making an image on SD.

**Comment by Brawlence** [comment]
Just out of curiosity, what's your PC specs? Do you have enough VRAM?

**Comment by punchewwastaken** [comment]
I basically followed this guide, the files may be a bit outdated but the command line flags are still correct as I have tested them and they work.
https://www.youtube.com/watch?v=G6sDNpcgrbc
